griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,400 @@
+from __future__ import annotations
+import logging
+from schema import Schema, Literal, Optional, Or
+from attr import define, field
+from griptape.artifacts import (
+    ErrorArtifact,
+    InfoArtifact,
+    ListArtifact,
+    BlobArtifact,
+)
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+
+
+@define
+class GoogleSheetsClient(BaseGoogleClient):
+    SHEETS_SCOPES = ["https://www.googleapis.com/auth/spreadsheets"]
+
+    DRIVE_READ_SCOPES = ["https://www.googleapis.com/auth/drive.readonly"]
+
+    DRIVE_AUTH_SCOPES = ["https://www.googleapis.com/auth/drive"]
+
+    DRIVE_UPLOAD_SCOPES = ["https://www.googleapis.com/auth/drive.file"]
+
+    DEFAULT_FOLDER_PATH = "root"
+
+    owner_email: str = field(kw_only=True)
+
+    @activity(
+        config={
+            "description": "Can be used to list all spreadsheets in the specified folder.",
+            "schema": Schema(
+                {
+                    Optional(
+                        "folder_path",
+                        default=DEFAULT_FOLDER_PATH,
+                        description="Path of the folder (like 'MainFolder/Subfolder1/Subfolder2') "
+                        "from which spreadsheets should be listed.",
+                    ): str,
+                }
+            ),
+        }
+    )
+    def list_spreadsheets(self, params: dict) -> ListArtifact | ErrorArtifact:
+        folder_path = params.get("folder_path", self.DEFAULT_FOLDER_PATH)
+
+        try:
+            service = self._build_client(
+                scopes=self.DRIVE_READ_SCOPES,
+                service_name="drive",
+                version="v3",
+                owner_email=self.owner_email,
+            )
+
+            if folder_path == self.DEFAULT_FOLDER_PATH:
+                query = "mimeType='application/vnd.google-apps.spreadsheet' and 'root' in parents and trashed=false"
+            else:
+                folder_id = self._convert_path_to_file_id(service, folder_path)
+                if folder_id:
+                    query = (
+                        f"'{folder_id}' in parents and mimeType='application/vnd.google-apps.spreadsheet' "
+                        f"and trashed=false"
+                    )
+                else:
+                    return ErrorArtifact(
+                        f"Could not find folder: {folder_path}"
+                    )
+
+            spreadsheets = []
+            page_token = None
+
+            while True:
```

Comment:
Change to `while page_token is not None`

Reply:
Done!